src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+        List<RassBusinessObjectUpdateResult<Proposal>> proposalResults = proposalResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(proposalResults, Proposal.class, false);
+    }
+
+    private void writeAwardsDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Award> awardResultGrouping = processingResults.getAwardResults();
+        List<RassBusinessObjectUpdateResult<Award>> awardResults = awardResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(awardResults, Award.class, true);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsDetailsToReport(List<RassBusinessObjectUpdateResult<R>> results, Class businessObjectClass,
+            boolean shouldReportUpdates) {
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SUCCESS_NEW);
+        if (shouldReportUpdates) {
+            writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SUCCESS_EDIT);
+        }
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.ERROR);
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SKIPPED);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsDetailsToReportByResultCode(List<RassBusinessObjectUpdateResult<R>> results,
+            Class businessObjectClass, RassObjectUpdateResultCode resultCode) {
+        List<RassBusinessObjectUpdateResult<R>> filteresdResults = results.stream().filter(result -> resultCode.equals(result.getResultCode()))
+                .collect(Collectors.toList());
+
+        writeHeaderForResultsDetailsByResultCode(businessObjectClass, resultCode);
+
+        for (RassBusinessObjectUpdateResult<R> result : filteresdResults) {
+            writeResultDetailToReport(result);
+        }
+    }
+
+    private void writeHeaderForResultsDetailsByResultCode(Class businessObjectClass, RassObjectUpdateResultCode resultCode) {
+        String header = StringUtils.EMPTY;
+        String objectName = businessObjectClass.getSimpleName();
+        switch (resultCode) {
+            case SUCCESS_NEW:
+                header = "** " + objectName + " objects successfully created:";
+                break;
+            case SUCCESS_EDIT:
+                header = "** " + objectName + " objects successfully updated:";
+                break;
+            case ERROR:
+                header = "** " + objectName + " objects that had errors:";
+                break;
+            case SKIPPED:
+                header = "** " + objectName + " objects that were skipped:";
+                break;
+        }
```

Comment:
Another switch block where it would be helpful to have a "default" case.

Reply:
added